api/src/main/java/io/strimzi/api/kafka/model/balancing/KafkaRebalanceMode.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.balancing;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public enum KafkaRebalanceMode {
+    FULL_REBALANCE,
+    ADD_BROKER,
+    REMOVE_BROKER;
+
+    @JsonCreator
+    public static KafkaRebalanceMode forValue(String value) {
+        switch (value) {
+            case "full-rebalance":
+                return FULL_REBALANCE;
+            case "add-broker":
+                return ADD_BROKER;
+            case "remove-broker":
+                return REMOVE_BROKER;
+            default:
+                return null;
```

Comment:
Maybe it would be better to return an `Optional<KafkaRebalanceMode>` from this method instead of `null`?

Reply:
It's mostly used for deserialization (as per @JsonCreator annotation) and it's not used in the code. Not sure how Jackson would deal with the Optional. I would leave null.